Adds a vscode devcontainer spec

daniel-simpson opened the pull request:
Plus some editor extensions to work with Go

Closes #215

reaper47 reviewed (changes requested):
The container is missing [task](https://taskfile.dev), [hugo extended](https://gohugo.io/about/what-is-hugo), nodejs/npm, and [templ](templ.guide) to build the software successfully with `task build`.

Here's a devcontainer.json that works:
```json
{
	"name": "Go",
	"build": {
		"dockerfile": "Dockerfile"
	},
	"forwardPorts": [
		8078
	],
	"postCreateCommand": "git submodule update --init && go mod download && go mod tidy",
	"customizations": {
		"vscode": {
			"extensions": [
				"ms-vscode.go",
				"davidanson.vscode-markdownlint",
				"shardulm94.trailing-spaces",
				"IBM.output-colorizer",
				"golang.go",
				"budparr.language-hugo-vscode",
				"a-h.templ"
			],
			"settings": {
				"go.useLanguageServer": true
			}
		}
	},
	"remoteUser": "root"
}
```

Here's the Dockerfile that sits right next to devcontainer.json:
```Dockerfile
FROM mcr.microsoft.com/devcontainers/go:1-1.21-bullseye

RUN apt-get update && apt-get install -y curl wget git

RUN curl -fsSL https://deb.nodesource.com/setup_current.x | bash - && \
    apt-get install -y nodejs \
    build-essential

RUN CGO_ENABLED=1 go install -tags extended github.com/gohugoio/hugo@latest
RUN go install github.com/go-task/task/v3/cmd/task@latest
RUN go install github.com/a-h/templ/cmd/templ@latest
```
The only problem remaining is running the executable fails:
```text
> ./bin/recipya s
Missing required env variable: RECIPYA_SERVER_PORT
Missing required env variable: RECIPYA_SERVER_URL
Application setup will terminate
```

It happens even when `config.json` exists because the setup [only considers](https://github.com/reaper47/recipya/blob/main/internal/app/setup.go#L131) environment variables when the app is ran from a Docker container. To fix this, we need to skip this `if` condition when such file exists.

daniel-simpson commented:
I've switched to using your devcontainer (a huge improvement!) and added:
* A `build` task to generate the documentation site and other runtime prerequisites
* A `Debug` launch.json configuration, which runs a recipya development instance at http://localhost:8079 and opens it once it's up and running.
 
Note that I chose not to make the `Build` task an explicit prerequisite of the `Debug` launch configuration, as it takes quite  a while to run, and isn't strictly necessary for subsequent runs.

Tested out the dev environment to fix up my last PR: #223.  Works really nicely!

daniel-simpson commented:
A few more tasks in here now, though looks like I've broken something in the build 😬.  I'll keep at it, sing out if you spot me diverging from what you'd prefer.

reaper47 reviewed (changes requested):
The container works like a charm. Here goes two minor things before the merge!

daniel-simpson commented on .vscode/tasks.json:
```diff
@@ -0,0 +1,70 @@
+                "i",
+            ]
+        },
+        {
+            "label": "Build Javascript",
+            "type": "shell",
+            "dependsOn": "Install NPM packages",
+            "command": "npm",
+            "args": [
+                "run",
+                "build"
+            ]
+        },
```
You're absolutely right.  I meant to add a dev mode with file watching for `npm run build` but didn't get around to it.

I'll remove these for now, and only add them back in if I figure out something that adds some value here.

daniel-simpson commented on .vscode/tasks.json:
```diff
@@ -0,0 +1,70 @@
+            "args": [
+                "i",
+            ]
+        },
```
Removed
